sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/AsyncReadBenchmark.java

```diff
@@ -78,34 +74,26 @@ protected void performWorkload(BaseSubscriber<PojoizedJson> baseSubscriber, long
 
     private void readLatency(Mono<PojoizedJson> readItem, BaseSubscriber<PojoizedJson> baseSubscriber, long i) {
         Mono sparsitySleepMono = sparsityMono(i);
-        Mono<PojoizedJson> result = readItem;
+        Mono<PojoizedJson> result;
         LatencySubscriber<PojoizedJson> latencySubscriber = new LatencySubscriber<>(baseSubscriber);
-        if (sparsitySleepMono != null) {
-            result = sparsitySleepMono.flux().flatMap(
-                null,
-                null,
-                () -> {
-                    latencySubscriber.context = latency.time();
-                    return readItem;
-                }).single();
-        } else {
-            latencySubscriber.context = latency.time();
-        }
+        result = sparsitySleepMono.flux().flatMap(
+            null,
+            null,
+            () -> {
+                latencySubscriber.context = latency.time();
+                return readItem;
+            }).single();
 
         result.subscribeOn(Schedulers.parallel()).subscribe(latencySubscriber);
     }
 
     private void readThroughput(Mono<PojoizedJson> readItem, BaseSubscriber<PojoizedJson> baseSubscriber, long i) {
         Mono sparsitySleepMono = sparsityMono(i);
-        Mono<PojoizedJson> result = readItem;
-        if (sparsitySleepMono != null) {
```

Comment:
the reason initially I decided to go with null value for `sparsitySleepMono` was because it would help to bypass the flatmap. I suspect you will have a ~1-2% perf hit as a result of this change on the normal path.

could you do a perf run using this benchmark tool both on the master and on your PR and measure the perf difference?